transition_lib/transition_api_lib.py

```diff
@@ -33,105 +35,82 @@ def build_headers():
             return error
         
     @staticmethod    
     def get_transition_routing_modes():
         try:
-            body = Transition.build_body()
-            result = requests.post(f"{Transition.DEFAULT_URL}routing-modes", json=body)
+            headers = Transition.build_headers()
+            result = requests.get(f"{Transition.API_URL}/routing-modes", headers=headers)
             if result.status_code == 200:
                 # Query returns the list as a string, so we need to parse it.
                 # Can probably be done differently
                 result = [x.replace("[", "").replace("]", "").replace('"', "") for x in result.text.split(",")]
                 return result
             else:
-                print(f"Routing modes request unsuccessfull: {result.status_code} {result.text}")
-                return f"Request unsuccessfull: {result.status_code} {result.text}"
+                return f"Request to /routing-modes unsuccessfull: {result.status_code} {result.text}"
         except requests.RequestException as error:
             return error
         
-    
-    # The following methods are not used with the current endpoint implementation,
-    # but will be used when tha API is updated to include the token.
-
-    # @staticmethod     
-    # def call_api_queries():
-    #     try:
-    #         headers = Transition.build_headers()
-    #         response = requests.post(f"{Transition.DEFAULT_URL}queries", headers=Transition.headers)
-    #         if response.status_code == 200:
-    #             print(f"Request successfull: {response.text}")
-    #             return f"Request successfull: {response.text}"
-    #         else:
-    #             print(f"Request unsuccessfull: {response.status_code} {response.text}")
-    #             return f"Request unsuccessfull: {response.status_code} {response.text}"
-    #     except requests.RequestException as e:
-    #         return f"Error: {e}"
-        
-    # @staticmethod
-    # def get_token():
-    #     try:
-    #         body = Transition.build_body()
-    #         response = requests.get(f"{Transition.DEFAULT_URL}token", json=body)
-    #         if response.status_code == 200:
-    #             # Update the token in the config file
-    #             config['credentials']['token'] = response.text
-    #             with open('config.ini', 'w') as configfile:
-    #                 config.write(configfile)
-    #             return f"Request successfull: {response.text}"
-    #         else:
-    #             print(f"Request unsuccessfull: {response.status_code} {response.text}")
-    #             return f"Request unsuccessfull: {response.status_code} {response.text}"
-    #     except requests.RequestException as e:
-    #         return f"Error: {e}"
+    @staticmethod
+    def get_token():
+        try:
+            body = Transition.build_body()
+            response = requests.post(f"{Transition.DEFAULT_URL}/token", json=body)
```

Comment:
nitpicking mais avoir un Transition.TOKEN_URL serait plus clair (??)

Reply:
selon moi ça avait plus de sens d'avoir le base URL et le API URL qui est baseURL/api
Je pourrais renommer les variables en BASE_URL et API_URL wdyt ?

Reply:
that works